src/librustc_driver/driver.rs

```diff
@@ -1275,9 +1275,9 @@ pub fn build_output_filenames(input: &Input,
                                            .values()
                                            .filter(|a| a.is_none())
                                            .count();
-            let ofile = if unnamed_output_types > 1 {
-                sess.warn("ignoring specified output filename because multiple outputs were \
-                           requested");
+            let ofile = if unnamed_output_types > 1 && sess.opts.output_types.get(&OutputType::Exe).is_none() {
+                sess.warn("warning: ignoring output name requested with -o for \"link\" output because
```

Comment:
This'll need a trailing `\` to ensure that the output doesn't show up on two lines